Some smaller fixes

s-scherrer opened the pull request:
- added pyproject.toml (#211)
- removed dependency on deprecated numpy API (#214)
- better calculation of moments in cython metrics
- updated PyScaffold to version 4
- use pytest directly in GH actions
- added pre-commit hooks for running flake8 on diff

wpreimes commented on docs/conf.py:
```diff
@@ -132,19 +143,19 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
-html_theme = 'sphinx_rtd_theme'
+html_theme = "alabaster"
```
I think we use the other template normally, I dont like that one ;)

wpreimes commented on setup.cfg:
```diff
@@ -99,6 +100,13 @@ markers =
     slow : marks slow tests (deselect with '-m "not slow"')
 testpaths = tests
+# This removes some of the warnings that show up with pytest but are not an issue
+filterwarnings =
```
:+1:

wpreimes commented on src/pytesmo/__init__.py:
```diff
@@ -19,4 +44,4 @@
 
 testdata_path = os.path.join(src_path, '..', 'tests', 'test-data')
```
I wonder if there is a better solution, in the tests it is nice to import the testdata path... but maybe this should not be here...

s-scherrer replied:
I think there is a better solution, but I have to look it up

s-scherrer replied:
testdata_path is only used in the examples anyways, I changed the examples accordingly and removed it from the package.

wpreimes commented on src/pytesmo/time_series/filters.pyx:
```diff
@@ -38,25 +38,16 @@ import platform
 @cython.cdivision(True)
-def exp_filter(np.ndarray[DTYPE_d, ndim=1] in_data,
-               np.ndarray[DTYPE_d, ndim=1] in_jd,
+def exp_filter(floating [:] in_data,
```
for the exponential filter we actually have another package (pyswi) which is not yet public... we should make pyswi public (currently on gitlab.geo.tuwien.ac.at) and then remove it from pytesmo. @sebhahn and @Adeaem were working on this

s-scherrer replied:
I only updated the existing code here to not depend on the deprecated numpy API anymore. I would suggest to postpone the removal to another PR, so I can merge this here in soon.

wpreimes replied:
sure, we should only remove it when pyswi is public

wpreimes commented on .github/workflows/ci.yml:
```diff
@@ -54,8 +54,8 @@ jobs:
           python setup.py install
-          python setup.py test --addopts "--cache-clear -m 'not full_framework'"
-          python setup.py test --addopts "--cache-clear -m 'full_framework'"
+          pytest --cache-clear -m 'not full_framework'
```
btw, this was the fix for avoiding the segfault. no idea where they came from...

wpreimes commented on tests/test_validation_framework/test_adapters.py:
```diff
@@ -12,191 +12,251 @@
-from datetime import datetime
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore")
+    from ascat.read_native.cdr import AscatSsmCdr
```
why is the ascat package throwing so many warnings anyway @sebhahn ?

wpreimes reviewed (changes requested):
Can you change the docs template please? Otherwise looks fine

wpreimes commented:
I wonder if you need to add `coverage` here https://github.com/s-scherrer/pytesmo/blob/916defdc284288b33fa2fb666d98b85f9408f825/setup.cfg#L65

s-scherrer commented on .coveragerc:
```diff
@@ -2,12 +2,6 @@
 source = pytesmo
-# omit = bad_file.py
-
-[paths]
-source =
-    src/
-    */site-packages/
```
This was causing the issue with coverage, @wpreimes

s-scherrer commented:
- added pyproject.toml (#211)
- removed dependency on deprecated numpy API (#214)
- better calculation of moments in cython metrics
- updated PyScaffold to version 4
- use pytest directly in GH actions
- added pre-commit hooks for running flake8 on diff
